qemu/tests/seabios.py

```diff
@@ -34,24 +35,104 @@ def boot_menu_check():
+
+            pattern = ""
+            if match_rhel6:
+                pattern += "bios.bin"
+            elif match_rhel7 or match_pc:
+                pattern += "bios-256k.bin"
+            else:
+                err_msg = "New machine type needs to check:   is %s " % m_type
+                test.error(err_msg)
+
+            env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+            vm = env.get_vm(params["main_vm"])
+            session = vm.wait_for_login(timeout=login_timeout)
+            output = vm.monitor.info("roms")
+            match_bin = re.search(pattern, output)
+            if match_bin:
+                print("Check seabios bin succeeded!")
+            else:
+                err_msg = "Check seabios bin faile when machine type is %s " % m_type
+                test.error(err_msg)
+            vm.monitor.cmd("system_reset")
+        return
+
+    if 'reboottimeout' in params:
+        rtime_cfg = int(params.get("reboottimeout"))
+        rtime_sec = rtime_cfg/1000
+        time.sleep(20)
+        if rtime_sec < 60:
+            time.sleep(60)
+        else:
+            time.sleep(rtime_sec)
+
+        output = seabios_session.get_stripped_output()
+        if 0 <= rtime_cfg <= 65535:
+            pattern = "No bootable device.  Retrying in %s seconds" % rtime_sec
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+        elif rtime_cfg > 65535:
+            pattern = "No bootable device. Retrying in 65 seconds"
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+        elif rtime_cfg < 0:
+            pattern1 = "No bootable device."
+            pattern2 = " Retrying in"
+            match1 = re.search(pattern1, output)
+            match2 = re.search(pattern2, output)
+            if match1 and not match2:
+                logging.info("Reboot timeout setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+    return
 
     timeout = float(params.get("login_timeout", 240))
     boot_menu_key = params.get("boot_menu_key", 'f12')
     restart_key = params.get("restart_key")
     boot_menu_hint = params.get("boot_menu_hint")
     boot_device = params.get("boot_device", "")
     sgabios_info = params.get("sgabios_info")
-
-    seabios_session = vm.logsessions['seabios']
+    logging.info("Start guest with sga bios")
 
     if sgabios_info:
-        error.context("Display and check the SGABIOS info", logging.info)
+        logging.info("Display and check the SGABIOS info")
```

Comment:
the same

Reply:
@suqinhuang 
Currently, @error.context_aware isn't recommended, But error.context depend on @error.context_aware.